qiskit/circuit/library/generalized_gates/permutation.py

```diff
@@ -75,24 +82,12 @@ def __init__(
         circuit = QuantumCircuit(num_qubits, name=name)
 
         super().__init__(num_qubits, name=name)
+
+        # pylint: disable=cyclic-import
+        from qiskit.synthesis.permutation.permutation_utils import _get_ordered_swap
```

Comment:
Should we drop the leading `_` here if we're going to be reusing it across multiple modules?

Reply:
This particular function feels like a utility function that only developers would use. In addition, this code will probably completely disappear from `Permutation` when we turn `Permutation` into a `Gate` instead of a `QuantumCircuit`. Having said that, it would probably eventually make sense to externalize a synthesis method/plugin based on this function (which would take a permutation pattern and return a quantum circuit).